akiuri/akiuri.go

```diff
@@ -12,19 +19,26 @@ const (
 type ObjectType int
 
 const (
-	UNKNOWN_TYPE ObjectType = iota
-	SPEC
-	TRACE // aka learn session
+	SPEC  ObjectType = iota
+	TRACE            // aka learn session
 )
 
-func stringToObjectType(s string) ObjectType {
+func (o ObjectType) Ptr() *ObjectType {
+	return &o
+}
+
+func (o1 *ObjectType) Is(o2 ObjectType) bool {
+	return o1 != nil && *o1 == o2
+}
```

Comment:
Another alternative is to have IsTrace() and IsSpec() methods, though Go stdlib has been moving more towards a model like this for I/O errors.  But right now your code in the other PR looks like

Is(akiuri.Trace)

instead of

IsTrace()